packages/dds/tree/src/feature-libraries/forest-summary/incrementalSummaryBuilder.ts

```diff
@@ -495,17 +529,26 @@ export class ForestIncrementalSummaryBuilder implements IncrementalEncoderDecode
 	}
 
 	/**
-	 * Called to get the encoded contents of an incremental chunk with the given reference ID.
-	 * This is typically used when loading the forest to retrieve the contents of incremental chunks.
-	 * @param referenceId - The reference ID of the chunk to retrieve.
-	 * @returns The encoded contents of the chunk.
+	 * {@link IncrementalEncoder.decodeIncrementalChunk}
 	 */
-	public getEncodedIncrementalChunk(referenceId: ChunkReferenceId): EncodedFieldBatch {
-		const chunkEncodedContents = this.encodedChunkContentsMap.get(`${referenceId}`);
-		assert(
-			chunkEncodedContents !== undefined,
-			0xc26 /* Incremental chunk contents not found */,
-		);
-		return chunkEncodedContents;
+	public decodeIncrementalChunk(
+		referenceId: ChunkReferenceId,
+		chunkDecoder: (encoded: EncodedFieldBatch) => TreeChunk,
+	): TreeChunk {
+		const ChunkLoadProperties = this.loadedChunksMap.get(`${referenceId}`);
+		assert(ChunkLoadProperties !== undefined, "Encoded incremental chunk not found");
+		const chunk = chunkDecoder(ChunkLoadProperties.contents);
+
+		// Track the decoded chunk. This will recreate the tracking state when the summary that this client
+		// is loaded from was generated. This is needed to ensure that incremental summaries work correctly
+		// when a new client starts to summarize.
+		setInNestedMap(this.chunkTrackingPropertiesMap, this.initialSequenceNumber, chunk, {
+			referenceId,
+			summaryPath: ChunkLoadProperties.summaryPath,
+		});
+
+		// Add a ref-count to this chunk similar to the one added when "encodeIncrementalField" is called.
+		chunk.referenceAdded();
```

Comment:
It doesn't actually matter here, but generally its good to add to a ref count before adding the reference its counting to help ensure things don't get disposed/modified inbetween.

Also I think this comment is more informative.


```suggestion
		// Account for the reference about to be added in chunkTrackingPropertiesMap
		// to ensure that no other users of this chunk think they have unique ownership.
		// This prevents prevent whoever this chunk is returned to from modifying it in-place.
		chunk.referenceAdded();

		// Track the decoded chunk. This will recreate the tracking state when the summary that this client
		// is loaded from was generated. This is needed to ensure that incremental summaries work correctly
		// when a new client starts to summarize.
		setInNestedMap(this.chunkTrackingPropertiesMap, this.initialSequenceNumber, chunk, {
			referenceId,
			summaryPath: ChunkLoadProperties.summaryPath,
		});
```